diff --git a/src/repos.rs b/src/repos.rs
index e20fa83..9b2ef2e 100644
--- a/src/repos.rs
+++ b/src/repos.rs
@@ -84,11 +84,21 @@ pub(crate) fn find_repos(
             }
             repos.insert_repo(name, repo);
         } else if file.path.is_dir() && file.depth > 0 {
-            let read_dir = fs::read_dir(file.path)
-                .change_context(TmsError::IoError)?
-                .map(|dir_entry| dir_entry.expect("Found non-valid utf8 path").path());
-            for dir in read_dir {
-                to_search.push_back(SearchDirectory::new(dir, file.depth - 1))
+            match fs::read_dir(&file.path) {
+                Err(ref e) if e.kind() == std::io::ErrorKind::PermissionDenied => {
+                    eprintln!(
+                        "Warning: no permission to read '{0}'",
+                        file.path.to_string()?
+                    );
+                }
+                r => {
+                    let read_dir = r
+                        .change_context(TmsError::IoError)?
+                        .map(|dir_entry| dir_entry.expect("Found non-valid utf8 path").path());
+                    for dir in read_dir {
+                        to_search.push_back(SearchDirectory::new(dir, file.depth - 1))
+                    }
+                }
             }
         }
     }
